compiler/destroyer.nim

```diff
@@ -127,20 +127,13 @@ type
     owner: PSym
     g: ControlFlowGraph
     jumpTargets: IntSet
-    tmpObj: PType
-    tmp: PSym
-    destroys, topLevelVars: PNode
-    toDropBit: Table[int, PSym]
+    topLevelVars: PNode
+    destroys: OrderedTable[int, tuple[dropBit: PSym, destroy_call: PNode]] 
```

Comment:
Can do that

Reply:
Thanks, and now since the 2nd read of a sink parameter is really bad, the compiler should either warn or error about this, just like it does in other cases when it decides whether to move or to copy. Perfectly consistent design.

Reply:
I can do that, could you please look into #9809 in return please.